### PR TITLE
Cpp library gluecode

### PR DESCRIPTION
**Pull request by gantech**
Improving robustness in C++ API and including installation instructions and sample installation and configuration scripts.

### CODE REVIEW

**Comment by gantech** [comment]
@sayerhs The spack based config script doesn't quite work with linking to the hdf5 libraries. Could you please help with this issue when you get the time. @jrood-nrel is quite familiar with my frustrations on this issue.

**Comment by sayerhs** [comment]
@gantech I have added a new commit to your `cppLibraryGlueCode` that should fix the linking issue when building `OpenFAST` against `spack` installed libraries. Also see the corresponding update to the [NaluSpack](https://github.com/NaluCFD/NaluSpack) `master` branch that will allow you to do `spack install openfast@develop+cxx` and get all dependencies in one shot. 

```
==> Installing openfast
==> cmake is already installed in /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/cmake-3.8.0-b753mbz7v5bodrfoodnq6pjejz7ehdsk
==> hdf5 is already installed in /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/hdf5-1.8.16-ee2nsbowdrvy4npv6kvjlnjz4rcphoxp
==> libxml2 is already installed in /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/libxml2-2.9.4-hlegrasbrhpmm3mnitrv7jztvhai4ysb
==> zlib is already installed in /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/zlib-1.2.11-dayoxcvooeyba7urrdphgeqd7gr6h7xr
==> openmpi is already installed in /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/openmpi-1.10.3-gp6gqlvjm25kf5n46ib5jkv3mfse377a
==> yaml-cpp is already installed in /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/yaml-cpp-0.5.3-jfzt747wz5mlc37lx6pdsiqi6tlor5uf
==> netlib-lapack is externally installed in /usr/lib64
==> Already fetched /home/sanantha/spack/var/spack/stage/openfast-develop-jedvy5ya5bdgtacr4uv7r74atyxqi5r4/openfast
==> No checksum needed when fetching with git
==> Already staged openfast-develop-jedvy5ya5bdgtacr4uv7r74atyxqi5r4 in /home/sanantha/spack/var/spack/stage/openfast-develop-jedvy5ya5bdgtacr4uv7r74atyxqi5r4
==> Already patched openfast
==> Building openfast [CMakePackage]
==> Executing phase : 'cmake'
==> Executing phase : 'build'
==> Executing phase : 'install'
==> Successfully installed openfast
  Fetch: 0.00s.  Build: 13.99s.  Total: 14.00s.
[+] /home/sanantha/spack/opt/spack/linux-centos6-x86_64/gcc-5.2.0/openfast-develop-jedvy5ya5bdgtacr4uv7r74atyxqi5r4
```

**Comment by gantech** [comment]
@rafaelmudafort and @sayerhs I updated the installation instructions. Could you provide some feedback on this. I mostly copied them from @jrood-nrel's instructions for Nalu.

File: docs/source/install_linux.rst [hunk 1]
```diff
@@ -51,11 +51,10 @@ Step 1
 ~~~~~~
 
 This assumes you have a Homebrew installation of GCC installed already 
-(we are using GCC 6.3.0). These instructions have been tested on OSX 10.11. MacOS 10.12 
-will not build CMake with GCC anymore, so these instructions won't work
```
Comment (gantech) [inline docs/source/install_linux.rst]: Thanks to @rafaelmudafort for pointing this line out. I'm not sure about this. Hence I'm taking this out.

**Comment by gantech** [comment]
Thanks for all the modifications to the installation instructions @rafaelmudafort. Could we merge this into dev then?

**Pull request merged**